### PR TITLE
PP-9669 Adds dispute reporting page, disputes endpoint API reference

### PR DESCRIPTION
**Pull request by NathanD-GDS**
# DO NOT MERGE - this will be merged when the endpoint goes live

### Context
GOV.UK Pay have added a new endpoint to our API - `v1/disputes`. This endpoint allows services to search for disputed payments to see their status, amount, and the reason for the dispute.

### Changes proposed in this pull request
This PR adds:

- a new page about reporting on disputes
- a new page to the API reference indexing the new endpoint
- updates mock card numbers for testing disputes
- fixes a few formatting errors

### CODE REVIEW

File: source/disputes/index.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    }
+  ]
+}
+
+```
+
+Each object in `results` represents a single dispute and contains:
+
+* the amount of the disputed payment in pence (`amount`)
+* when the user raised the dispute (`created_date`)
```
Comment (stephencdaly) [inline source/disputes/index.html.md.erb:60]: I imagine there could be a delay between the user raising the dispute and us getting told about it due to delays with the bank processing it and sending it to Stripe - although I don't know about this. Would it be more accurate to say this is when the bank informed us that the user had raised a dispute? Not sure how much that accuracy matters here though.

File: source/disputes/index.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+| `product_not_received`  | The paying user claims they did not receive the products or services they purchased.                                                                                              |
+| `product_unacceptable`   | The paying user claims the service they paid for was poor or not described accurately before they paid for it.                                                                    |
+| `unrecognised`            | The paying user claims they do not recognise the payment on their bank statement.                                                                                                 |
+| <nobr>`subscription_cancelled` | The paying user claims you continued to charge them after they cancelled their subscription to your service.                                                                      |
+| `other`                   | Your payment service provider has given an unexpected dispute reason.<br><br>[Contact us and we’ll investigate](/support_contact_and_more_information/#report-an-incident). |
```
Comment (stephencdaly) [inline source/disputes/index.html.md.erb:95]: This links to `#report-an-incident`, but should it just link to the root of the support page?

File: source/api_reference/search_disputes_reference/index.html.md.erb [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+## Attributes you'll get in a 'Search disputes' response
+
+| Name                                  | Type           | Description                                |
+| ------------------------------------- | --------------- | ----------------------------------------- |
+| `page`                              | number          | The page of results you’re viewing.<br><br>To view other pages, make this request again using the `page` parameter.|
+| `count`                             | number          | Number of disputes on the current page of search results.|
+| `total`                             | number          | Number of total disputes matching your search criteria.|
+| `results`                           | object          | Contains disputes matching your search criteria.|
+| `amount`                            | number          | The amount of the disputed payment in pence.|
```
[the remark this answers is not in the document]
Comment (NathanD-GDS) [reply source/api_reference/search_disputes_reference/index.html.md.erb:115]: Good call. Fixing this in a new commit.

File: source/api_reference/search_disputes_reference/index.html.md.erb [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+| `created_date`                     | date (ISO 8601) | The date and time the paying user disputed this payment.|
+| `dispute_id`                       | string          | The unique ID GOV.UK Pay automatically associated with this dispute when the paying user disputed the payment.|
+| `evidence_due_date`               | date (ISO 8601) | The deadline for submitting your supporting evidence.<br><br>This value uses Coordinated Universal Time (UTC) and ISO 8601 format - `YYYY-MM-DDTHH:MM:SSZ`.|
+| `fee`                               | number          | The payment service provider’s dispute fee, in pence.|
+| `net_amount`                       | number          | The amount, in pence, your payment service provider (PSP) will adjust your next payout by.<br><br>For example, a `net_amount` of `-1500` means your PSP will take £15.00 from your next payout into your bank account.<br><br>`net_amount` is always a negative value.<br><br>`net_amount` only appears if you lose the dispute.|
```
[the remark this answers is not in the document]
Comment (NathanD-GDS) [reply source/api_reference/search_disputes_reference/index.html.md.erb:120]: How about:

> The amount, in pence, your payment service provider (PSP) will adjust your payout by, if they have not already adjusted a previous payout.

@stephencdaly

File: source/api_reference/search_disputes_reference/index.html.md.erb [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+| `reason`                            | string          | The reason the paying user gave for disputing this payment.<br><br>Possible values are:<br><br><li>`credit_not_processed`<br><li>`duplicate`<br><li>`fraudulent`<br><li>`general`<br><li>`product_not_received`<br><li>`product_unacceptable`<br><li>`unrecognised`<br><li>`subscription_cancelled`<br><li>`other`<br><br>You can [read about the meanings of the different `reason` values in our reporting guidance](/disputes/#reasons-for-disputes). |
+| `status`                            | string          | The current status of the dispute.<br><br>Possible values are:<br><br><li>`needs_response`<br><li>`won`<br><li>`lost`<br><li>`under_review`<br><br>You can [read more about the meanings of the different `status` values in our reporting guidance](/disputes/#dispute-status).|
+| <nobr>`settlement_summary`               | object          | Contains information about when a lost dispute was settled.<br><br>A dispute is settled when your payment service provider takes it from a payout to your bank account.<br><br>If you won a dispute, `settlement_summary` will be empty.|
```
[the remark this answers is not in the document]
Comment (NathanD-GDS) [reply source/api_reference/search_disputes_reference/index.html.md.erb:124]: Fixing in the next commit

File: source/disputes/index.html.md.erb [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+# Report on a dispute
+
+A dispute is when a paying user challenges a completed payment through their bank.
+
+There are many possible reasons for a dispute. For example, your user might not recognise a payment on their bank statement, or they might believe they’ve paid for your service twice.
+
+GOV.UK Pay will email you whenever a user disputes a payment. Our email will explain the reason for the dispute and what you need to do. You might need to provide evidence that the user needed to make the payment.
+
+If you lose a dispute, you must pay back the transaction amount and a dispute fee set by your payment service provider. You’ll only pay for lost disputes raised on or after 8 August 2022. If you lose a dispute that was raised before 8 August 2022, you will not be charged.
```
[the remark this answers is not in the document]
Comment (NathanD-GDS) [reply source/disputes/index.html.md.erb:16]: I think you're right - it sounds quite active. How about:

> If you lose a dispute, your payment service provider automatically takes the disputed amount and a dispute fee from the next payment into your bank account.

@stephencdaly

File: source/disputes/index.html.md.erb [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+      }
+    }
+  ]
+}
+
+```
+
+Each object in `results` represents a single dispute and contains:
+
+* the amount of the disputed payment in pence (`amount`)
```
[the remark this answers is not in the document]
Comment (NathanD-GDS) [reply source/disputes/index.html.md.erb:59]: Fixing in the next commit.

File: source/disputes/index.html.md.erb (hunk 2 above)
Comment (NathanD-GDS) [reply source/disputes/index.html.md.erb:95]: It should, yeah. No idea how that happened! Fixing in the next commit.

File: source/disputes/index.html.md.erb (hunk 6 above)
[the remark this answers is not in the document]
Comment (stephencdaly) [reply source/disputes/index.html.md.erb:16]: Yep sounds good

File: source/api_reference/search_disputes_reference/index.html.md.erb (hunk 4 above)
[the remark this answers is not in the document]
Comment (NathanD-GDS) [reply source/api_reference/search_disputes_reference/index.html.md.erb:120]: No, I agree - it's a bit awkward. I think splitting it into two sentences is a good shout. I'll go with:

> The amount, in pence, your payment service provider will take for a lost dispute. `net_amount` is deducted from your payout after you lose the dispute.

[the remark this answers is not in the document]
Comment (stephencdaly) [reply source/api_reference/search_disputes_reference/index.html.md.erb:120]: Yeh sounds good!

File: source/disputes/index.html.md.erb (hunk 1 above)
Comment (NathanD-GDS) [reply source/disputes/index.html.md.erb:60]: I think it's worth clarifying. I've changed this to:

> The date and time the user's bank told GOV.UK Pay about this dispute.

**Review by stephencdaly — APPROVED** [review]
All looks good!

File: source/api_reference/search_disputes_reference/index.html.md.erb [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+| <nobr>`from_settled_date` | date (ISO 8601) | Returns disputes settled on or after the `from_settled_date`.<br><br>Date must be in ISO 8601 format to date-level accuracy - `YYYY-MM-DD`.<br><br>Disputes are settled when your payment service provider takes the disputed amount from a payout to your bank account.<br><br>You can only use `from_settled_date` to find payments raised on or after 8 August 2022. |
+| `to_settled_date`   | date (ISO 8601) | Returns disputes settled before the `to_settled_date`.<br><br>Date must be in ISO 8601 format to date-level accuracy - `YYYY-MM-DD`.<br><br>Disputes are settled when your payment service provider takes the disputed amount from a payout to your bank account.<br><br>You can only use `to_settled_date` to find payments raised on or after 8 August 2022.          |
```
Comment (stephencdaly) [inline source/api_reference/search_disputes_reference/index.html.md.erb:41]:
```suggestion
| `to_settled_date`   | date (ISO 8601) | Returns disputes settled before the `to_settled_date`.<br><br>Date must be in ISO 8601 format to date-level accuracy - `YYYY-MM-DD`.<br><br>Disputes are settled when your payment service provider takes the disputed amount from a payout to your bank account.<br><br>You can only use `to_settled_date` to find disputes raised on or after 8 August 2022.          |
```

File: source/api_reference/search_disputes_reference/index.html.md.erb [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+| `status`              | string          | Returns disputes with a matching `status`.<br><br>`status` reflects what stage of the dispute process a dispute is at. <br><br>Available `status` values are:<br><br><li>`needs_response`<br><li>`won`<br><li>`lost`<br><li>`under_review`<br><br>[You can read more about the meanings of different `status` values](/disputes/#dispute-status).         |
+| <nobr>`from_settled_date` | date (ISO 8601) | Returns disputes settled on or after the `from_settled_date`.<br><br>Date must be in ISO 8601 format to date-level accuracy - `YYYY-MM-DD`.<br><br>Disputes are settled when your payment service provider takes the disputed amount from a payout to your bank account.<br><br>You can only use `from_settled_date` to find payments raised on or after 8 August 2022. |
```
Comment (stephencdaly) [inline source/api_reference/search_disputes_reference/index.html.md.erb:40]: Ah sorry the same change is needed on this line too
```suggestion
| <nobr>`from_settled_date` | date (ISO 8601) | Returns disputes settled on or after the `from_settled_date`.<br><br>Date must be in ISO 8601 format to date-level accuracy - `YYYY-MM-DD`.<br><br>Disputes are settled when your payment service provider takes the disputed amount from a payout to your bank account.<br><br>You can only use `from_settled_date` to find disputes raised on or after 8 August 2022. |
```

**Review by stephencdaly — APPROVED** [review]

**Pull request merged**